src/app/clusters/resource-monitoring-server/resource-monitoring-server.cpp

```diff
@@ -0,0 +1,356 @@
+}
+
+chip::Protocols::InteractionModel::Status Instance::UpdateCondition(uint8_t aNewCondition)
+{
+    auto oldCondition = mCondition;
+    mCondition        = aNewCondition;
+    if (mCondition != oldCondition)
+    {
+        MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::Condition::Id);
+    }
+    return Protocols::InteractionModel::Status::Success;
+}
+
+chip::Protocols::InteractionModel::Status
+Instance::UpdateChangeIndication(chip::app::Clusters::ResourceMonitoring::ChangeIndicationEnum aNewChangeIndication)
+{
+    if (aNewChangeIndication == chip::app::Clusters::ResourceMonitoring::ChangeIndicationEnum::kWarning)
+    {
+        if (!HasFeature(ResourceMonitoring::Feature::kWarning))
+        {
+            return Protocols::InteractionModel::Status::InvalidValue;
+        }
+    }
+    auto oldChangeIndication = mChangeIndication;
+    mChangeIndication        = aNewChangeIndication;
+    if (mChangeIndication != oldChangeIndication)
+    {
+        MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::ChangeIndication::Id);
+    }
+    return Protocols::InteractionModel::Status::Success;
+}
+
+chip::Protocols::InteractionModel::Status Instance::UpdateInPlaceIndicator(bool aNewInPlaceIndicator)
+{
+    auto oldInPlaceIndicator = mInPlaceIndicator;
+    mInPlaceIndicator        = aNewInPlaceIndicator;
+    if (mInPlaceIndicator != oldInPlaceIndicator)
+    {
+        MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::InPlaceIndicator::Id);
+    }
+    return Protocols::InteractionModel::Status::Success;
+}
+
+chip::Protocols::InteractionModel::Status Instance::UpdateLastChangedTime(DataModel::Nullable<uint32_t> aNewLastChangedTime)
+{
+    auto oldLastchangedTime = mLastChangedTime;
+    mLastChangedTime        = aNewLastChangedTime;
+    if (mLastChangedTime != oldLastchangedTime)
+    {
+        chip::app::GetAttributePersistenceProvider()->WriteScalarValue(
+            ConcreteAttributePath(mEndpointId, mClusterId, Attributes::LastChangedTime::Id), mLastChangedTime);
+        MatterReportingAttributeChangeCallback(mEndpointId, mClusterId, Attributes::LastChangedTime::Id);
+    }
+    return Protocols::InteractionModel::Status::Success;
+}
+
+uint8_t Instance::GetCondition() const
+{
+    return mCondition;
+}
+chip::app::Clusters::ResourceMonitoring::ChangeIndicationEnum Instance::GetChangeIndication() const
+{
+    return mChangeIndication;
+}
+
+chip::app::Clusters::ResourceMonitoring::DegradationDirectionEnum Instance::GetDegradationDirection() const
+{
+    return mDegradationDirection;
+}
+
+bool Instance::GetInPlaceIndicator() const
+{
+    return mInPlaceIndicator;
+}
+
+DataModel::Nullable<uint32_t> Instance::GetLastChangedTime() const
+{
+    return mLastChangedTime;
+}
+
+Status Instance::OnResetCondition()
+{
+    ChipLogDetail(Zcl, "ResourceMonitoringServer::OnResetCondition()");
+
+    // call apllication specific pre reset logic,
```

Comment:
^^ Fixing suggestion from above :) 

```suggestion
    // call application specific pre reset logic,
```